### PR TITLE
solution was added

### PR DESCRIPTION
**Pull request by Maksym845**
- [DEMO LINK](https://Maksym845.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

**Review by Oleksandra-Shevchenko — APPROVED** [review]
LGTM 👍

File: src/GoodsList.tsx [hunk 1]
```diff
@@ -8,7 +8,15 @@ type Props = {
 export const GoodsList: React.FC<Props> = ({ goods }) => (
   <ul>
     {goods.map(good => (
```
Comment (sergii-nosachenko) [inline src/GoodsList.tsx:10]: Can be destructurized

File: src/App.tsx [hunk 2]
```diff
@@ -1,27 +1,63 @@
...
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [users, setUsers] = useState<Good[] | null>(null);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const getGoods = async (request: () => Promise<Good[]>) => {
+    const response = await request().then(apiGoods => apiGoods);
```
Comment (sergii-nosachenko) [inline src/App.tsx:12]:
```suggestion
    const goods = await request();
```
You only need to wait result.

File: src/App.tsx [hunk 3]
```diff
@@ -1,27 +1,63 @@
...
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      return;
+    }
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    setUsers(null);
```
Comment (sergii-nosachenko) [inline src/App.tsx:22]: You can simply call `setUsers(goods || null);`

File: src/App.tsx [hunk 4]
```diff
@@ -1,27 +1,63 @@
...
-// or
-// import * as goodsAPI from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [users, setUsers] = useState<Good[] | null>(null);
```
Comment (sergii-nosachenko) [inline src/App.tsx:9]: Why users?

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/api/goods.ts [hunk 5]
```diff
@@ -10,10 +10,12 @@ export function getAll(): Promise<Good[]> {
 
 export const get5First = () => {
   return getAll()
-    .then(goods => goods); // sort and get the first 5
+    .then(goods => goods
```
Comment (Andr1yK) [inline src/api/goods.ts:13]: let's use `async await` :)

File: src/App.tsx [hunk 6]
```diff
@@ -1,27 +1,55 @@
...
-// or
-// import * as goodsAPI from './api/goods';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[] | null>(null);
```
Comment (Andr1yK) [inline src/App.tsx:9]: Why do u need this `null`? u can use empty array

File: src/GoodsList.tsx [hunk 7]
```diff
@@ -7,10 +7,22 @@ type Props = {
...
+
+      return (
+        <li
+          key={id}
+          data-cy="good"
+          style={
+            {
+              color,
+            }
+          }
```
Comment (Andr1yK) [inline src/GoodsList.tsx:21]:
```suggestion
          style={{  color }}
```

**Review by Andr1yK — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 6 above)
Comment (Maksym845) [reply src/App.tsx:9]: Denys said that null will be better... But anyway I will fix it.

Comment (Andr1yK) [reply src/App.tsx]: It's only about Objects (not about arrays)

File: src/api/goods.ts [hunk 8]
```diff
@@ -8,11 +8,11 @@ export function getAll(): Promise<Good[]> {
...
-      .sort((a, b) => a.name.localeCompare(b.name))
-      .slice(0, 5));
+export const get5First = async () => {
+  const goodsFromApi = await getAll();
+
+  return goodsFromApi.sort((a, b) => a.name.localeCompare(b.name))
+    .slice(0, 5);
 };
 
 export const getRedGoods = () => {
```
Comment (Andr1yK) [inline src/api/goods.ts:18]: Why didn't you rewrite this method?

File: src/App.tsx [hunk 9]
```diff
@@ -48,7 +48,7 @@ export const App: React.FC = () => {
         Load red goods
       </button>
 
-      {goods !== null
+      {goods
         && <GoodsList goods={goods} />}
```
Comment (Andr1yK) [inline src/App.tsx:52]: U can write it inline

{goods && <GoodsList goods={goods} />}

**Review by Andr1yK — CHANGES_REQUESTED** [review]

**Pull request closed without merging**